pkg/util/kafka/common.go

```diff
@@ -166,3 +171,41 @@ func newListenerSecurityProtocolMap(s string) listenerSecurityProtocolMap {
 
 	return listenerSecProtoMap
 }
+
+func GetBootstrapServers(cr *v1beta1.KafkaCluster) (bootstrapServers string) {
+	var listenerName string
+	var listenerServiceName string
+	var listenerStatusList v1beta1.ListenerStatusList
+
+	for _, lc := range cr.Spec.ListenersConfig.InternalListeners {
+		if lc.UsedForInnerBrokerCommunication && !lc.UsedForControllerCommunication {
+			listenerName = lc.Name
+		}
+	}
+
+	listenerStatusList, ok := cr.Status.ListenerStatuses.InternalListeners[listenerName]
+	if !ok {
+		return
+	}
+
+	switch {
+	case cr.Spec.HeadlessServiceEnabled:
+		listenerServiceName = HeadlessListenerServiceName
+	default:
+		listenerServiceName = AnyBrokerListenerServiceName
+	}
+
+	var bootstrapServersList []string
+
+	if len(listenerStatusList) > 0 {
+		for _, ls := range listenerStatusList {
+			if ls.Name == listenerServiceName {
+				bootstrapServers = ls.Address
+				return
```

Comment:
Not sure about using naked return in this func, its rather long for that and just creates confusion in general.

Reply:
Agreed. Will refactor.